SC_redesign/Crop_and_Soil/soil/percolation.py

```diff
@@ -0,0 +1,145 @@
+from typing import List, Optional
+from math import exp
+
+from SC_redesign.Crop_and_Soil.soil.layer_data import LayerData
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+"""
+This module is based on the section 'Percolation' (2:3.2) in SWAT
+"""
+
+
+class Percolation:
+    def __init__(self, soil_data: Optional[SoilData] = None):
+        self.data = soil_data or SoilData()
+
+    def percolate(self, has_seasonal_high_water_table: bool) -> None:
+        """executes percolation of excess water in each layer of soil profile to the layer directly beneath it
+
+        Args:
+            has_seasonal_high_water_table: if the HRU has a seasonal high water table (true/false)
+
+        SWAT Reference: sections 2:3.1 and 2
+        """
+        for i in range(len(self.data.soil_layers)):
+            """iterate through each layer of soil in the soil profile"""
+            upper_layer = self.data.soil_layers[i]
+            if i != (len(self.data.soil_layers) - 1):
+                """upper layer is not the bottom layer of the soil profile"""
```

Comment:
Can we reserve quote comments for docstrings or multi-line standalone comment/note blocks?

```suggestion
        for i in range(len(self.data.soil_layers)):  # iterate through each layer of soil in the soil profile
            upper_layer = self.data.soil_layers[i]
            if i != (len(self.data.soil_layers) - 1):  # upper layer is not the very bottom layer of the soil profile
```

Reply:
Yes. Are quote comments the ones that are going to be automatically put in the documentation?

Reply:
I believe all instances of this have now been corrected